packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -163,9 +163,21 @@ export interface AllowedTypeMetadata {
 	 */
 	readonly custom?: unknown;
 
-	// TODO metadata for enablable types will be added here
+	/**
+	 * If defined, indicates that an allowed type is enablable. Before upgrade, this type can be loaded into the tree from a document
+	 * but not written to the tree.
+	 */
+	readonly enabledUponSchemaUpgrade?: SchemaUpgradeToken;
 }
 
+/**
+ * Unique token used to upgrade schemas and determine if a particular upgrade has been completed.
+ *
+ * TODO:#38722 make this a unique type and use it for runtime schema upgrades
+ * @alpha
+ */
+export type SchemaUpgradeToken = true;
```

Comment:
Since this is leaking into the exported API surface, we should probably just be using a true placeholder. It being a Boolean makes it seem like a flag and not a token.

Since we don't want users making instances of this, using a sealed class which is only type exported and has a private constructor (use the trick FieldSchema uses) seems like the best approach here.